ui/src/components/content/catalog/Catalog.tsx

```diff
@@ -0,0 +1,70 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ */
+
+import React, { useEffect, useState } from 'react';
+import '../../../styles/catalog.css';
+import { DataNode } from 'antd/es/tree';
+import ServiceTree from './services/ServiceTree';
+import ServiceProvider from './services/ServiceProvider';
+import { BankOutlined } from '@ant-design/icons';
+import { useLocation } from 'react-router-dom';
+import { serviceVendorApi } from '../../../xpanse-api/xpanseRestApiClient';
+import { CategoryOclVo, VersionOclVo } from '../../../xpanse-api/generated';
+
+function Catalog(): JSX.Element {
+    const [key, setKey] = useState<string>('');
+    const [treeData, setTreeData] = useState<DataNode[]>([]);
+    const [categoryOclData, setCategoryOclData] = useState<CategoryOclVo[]>([]);
+    const location = useLocation();
+
+    useEffect(() => {
+        const path = location.hash.split('#')[1];
+        if (!path) {
+            return;
+        }
+        serviceVendorApi.listRegisteredServicesTree(path).then((rsp) => {
+            const data = rsp;
```

Comment:
do we need to assign it to a new variable here?

Reply:
fixed.